flink-read-write-hdfs/main.py

```diff
@@ -0,0 +1,148 @@
+        .column("item_id", types.String)
+        .column("item_count", types.Int32)
+        .column("timestamp", types.String)
+        .build()
+    )
+
+    purchase_events_source = FileSystemSource(
+        name="purchase_events",
+        path="/tmp/data/purchase_events.json",
+        data_format="json",
+        schema=purchase_events_schema,
+        timestamp_field="timestamp",
+        timestamp_format="%Y-%m-%d %H:%M:%S",
+    )
+
+    item_price_events_schema = (
+        Schema.new_builder()
+        .column("item_id", types.String)
+        .column("price", types.Float32)
+        .column("timestamp", types.String)
+        .build()
+    )
+
+    item_price_events_source = FileSystemSource(
+        name="item_price_events",
+        path="/tmp/data/item_price_events.json",
+        data_format="json",
+        schema=item_price_events_schema,
+        keys=["item_id"],
+        timestamp_field="timestamp",
+        timestamp_format="%Y-%m-%d %H:%M:%S",
+    )
+
+    # The total cost of purchases made by this user in the last 2 minutes.
+    f_total_payment_last_two_minutes = Feature(
+        name="total_payment_last_two_minutes",
+        dtype=types.Float32,
+        transform=OverWindowTransform(
+            expr="item_count * price",
+            agg_func="SUM",
+            window_size=timedelta(minutes=2),
+            group_by_keys=["user_id"],
+        ),
+    )
+
+    purchase_events_with_features = DerivedFeatureView(
+        name="purchase_events_with_features",
+        source=purchase_events_source,
+        features=[
+            "item_price_events.price",
+            f_total_payment_last_two_minutes,
+        ],
+        keep_source_fields=True,
+    )
+
+    client.build_features(
+        [
+            item_price_events_source,
+            purchase_events_with_features,
+        ]
+    )
+
+    result_table = client.get_features(purchase_events_with_features)
+
+    hdfs_sink = FileSystemSink(path="hdfs://namenode:8020/tmp/data/output", data_format="csv")
+
+    result_table.execute_insert(sink=hdfs_sink, allow_overwrite=True).wait()
+
+    purchase_events_with_features_schema = (
+        Schema.new_builder()
+            .column("user_id", types.String)
+            .column("item_id", types.String)
+            .column("item_count", types.Int32)
+            .column("timestamp", types.String)
+            .column("price", types.Float32)
+            .column("total_payment_last_two_minutes", types.Float32)
+            .build()
+    )
+
+    purchase_events_with_features_source = FileSystemSource(
+        name="purchase_events_with_features",
+        path="hdfs://namenode:8020/tmp/data/output",
+        data_format="csv",
+        schema=purchase_events_with_features_schema,
+        timestamp_field="timestamp",
+        timestamp_format="%Y-%m-%d %H:%M:%S",
+    )
+
+    saved_table = client.get_features(purchase_events_with_features_source)
+
+    saved_table_df = saved_table.to_pandas()
```

Comment:
Can we also print the output to a local file so that we can programmatically compare the output with the expected output?